### ISSUE TITLE
[cc_common] Linking output paths repeat parent directory of LibraryToLink

### ISSUE THREAD

**Issue by jjennebzl** [bazelbuild/bazel#13883]
### Description of the problem:

When passing a `name`  in `cc_common.link` to create a dynamic library, the generated file in the linking output's `LibraryToLink` has the parent directory's name repeated twice before the file name at the end. This is the case for both the dynamic library path that has been replaced into the `_S` form and the resolved symlink path too.

### Feature requests: what underlying problem are you trying to solve with this feature?

Stop repeating the parent directory name in the dynamic library output.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run `touch WORKSPACE`

In file `BUILD`:
```bazel
load("//:test.bzl", "elf")
elf(
    name = "my/long/name",
)
```

In file `test.bzl`
```bazel
load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")

def _impl(ctx):
    toolchain = find_cpp_toolchain(ctx)
    feature_config = cc_common.configure_features(
        ctx = ctx,
        cc_toolchain = toolchain,
    )

    linking_output = cc_common.link(
        actions = ctx.actions,
        feature_configuration = feature_config,
        cc_toolchain = toolchain,
        output_type = "dynamic_library",
        name = ctx.attr.name,
    )

    print(linking_output.library_to_link.resolved_symlink_dynamic_library)
    print(linking_output.library_to_link.dynamic_library)

elf = rule(
    implementation = _impl,
    attrs = {
        "_cc_toolchain": attr.label(
            default = Label("@bazel_tools//tools/cpp:current_cc_toolchain"),
        ),
    },
    fragments = ["cpp"],
)
```

Run `$ bazel build //:my/long/name`

In console output:
```log
...
DEBUG: /home/.../test.bzl:19:10: <generated file my/long/my/long/libname.so>
DEBUG: /home/.../test.bzl:19:10: <generated file _solib_k8/libmy_Slong_Smy_Slong_Slibname.so>
...
```

Here what I expect to see is: `<generated file my/long/libname.so>`.

### What operating system are you running Bazel on?

Red Hat Enterprise Linux Server release 7.9

### What's the output of `bazel info release`?

`release 4.1.0- (@non-git)`

###  Have you found anything relevant by searching the web?

No.

Thanks for any help/guidance on this issue!

**Comment by keith** [comment]
Fixed on master by b4cc44c978bc0e25b6652b66018b6aad12bff820 can you try with HEAD to verify this case?

**Comment by jjennebzl** [comment]
Yes, I've pulled, rebuilt and HEAD has fixed the problem - thanks for your help!
